zephyr/test/vmh.c

```diff
@@ -58,26 +58,37 @@ static void test_vmh_alloc_free_no_check(struct vmh_heap *heap,
 	}
 }
 
-static void verify_memory_content(void *ptr, uint32_t alloc_size)
+/* Fill memory with test pattern */
+static void verify_memory_content(void *ptr, uint32_t alloc_size, bool fill)
 {
+	const uint32_t pattern1 = (uint32_t)ptr ^ 0xDEADBEEF;
+	const uint32_t pattern2 = (uint32_t)ptr ^ 0xCAFEBABE;
+	const uint32_t pattern3 = (uint32_t)ptr ^ 0xFEEDFACE;
```

Comment:
hmm, just from the rigor PoV: your "random" XOR masks are all "letters," i.e. all in the range of 9 < x < 16, which also means that the high bit is always set, and since the high bit in pointers on Intel ADSP is also always set (0x9..., 0xa..., 0xb...) except for uncached 0x4... addresses on some platforms, that bit becomes not random at all :-) Maybe using more random patterns would be better.

Reply:
Are those patterns needed at all? We need only to check if the mapping went ok we are not checking anything else. That is why there was simple write/read check.

Reply:
@lyakh @dabekjakub  The purpose of these tests is not to verify the correct operation of memory hardware. I don't think it matters what bits are set. The idea was just to use 3 different tags that would be unique for each buffer. Thanks to this, in addition to checking that the memory has been correctly remapped, we also check that it has not been modified by other allocations.